Kudu.Services/ServiceHookHandlers/FetchHandler.cs

```diff
@@ -118,6 +118,7 @@ public class FetchHandler : HttpTaskAsyncHandler
                         context.ApplicationInstance.CompleteRequest();
                         break;
                     case FetchDeploymentRequestResult.RanSynchronously:
+                    case FetchDeploymentRequestResult.RanSynchronouslyFailed:
```

Comment:
This is actually the reason I left the feature in question out of Zipdeploy - because synchronous fetch deployment didn't do it. I didn't just want to add it and change the expected behavior, and it made more sense (to me, at the time) to keep the behavior the same between fetch and zip/push, so I did. @davidebbo what's your take on this?

Reply:
So is the /api/zip behavior today that if something goes wrong on the server, the request doesn't fail? If so, that doesn't seem good, and I wouldn't make push zip behave the same way (ideally we'd fix both).

Reply:
I changed fetch to act the same way as push on RanSynchronously, but I'm not 100% sure of the consequences here so feedback is appreciated. Tests seems to run fine at least 💯

Reply:
Not /api/zip, but /deploy (fetch deployment), which zipdeploy has the most in common with (a zip deployment just pushes the zip to the server and then does a "fetch" deployment with a custom fetch action that unzips the local file and sets the deployment source to the extract directory).

In DeploymentManager.Build(), the call to builder.Build() (which does most of the actual work of the deployment) is wrapped in a try/catch that doesn't rethrow. See https://github.com/projectkudu/kudu/blob/master/Kudu.Core/Deployment/DeploymentManager.cs#L651-L675. Since nothing further up the call stack checks the status, you get a 200 back. It's always been this way.

Reply:
Ah I see. I guess the reason is that in those scenarios, the http call normally comes from GitHub/BitBucket, and the error is less relevant. Instead, the deployment status shows it. But for Zip, I can see how not having an error becomes more quirky.